flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/operators/join/stream/StreamingJoinOperator.java

```diff
@@ -186,17 +189,22 @@ public void processElement1(StreamRecord<RowData> element) throws Exception {
                 // performance optimization: if the input key is null and it's
                 // not an outer side, we can simply ignore the input row
                 RowDataStringSerializer rowStringSerializer = new RowDataStringSerializer(leftType);
-                LOG.debug("dropping LEFT input {}", rowStringSerializer.asString(element.getValue()));
+                LOG.debug("Performance Optimization - Dropping left input {}", rowStringSerializer.asString(element.getValue()));
                 leftInputDroppedNullKeyCount.inc();
                 return;
-            } else if (isEquijoin && !isBatchMode() && statelessNullKeysEnabled) {
+        } else if (StreamingPerfOptimizationUtil.isEligibleForNullEquijoinOptimization(
+                    true,
+                    isEquijoin,
+                    isBatchMode(),
+                    statelessNullKeysEnabled,
+                    true)) {
                 // special optimization for outer joins on a NULL key
                 // we do not need to maintain any state since the join
                 // condition will result in an association failure, so
                 // we will never have associated rows data from the
                 // other side to pass through
-                // RowDataStringSerializer rowStringSerializer = new RowDataStringSerializer(leftType);
-                // LOG.info("SERGEI left input optimization {}", rowStringSerializer.asString(element.getValue()));
+                RowDataStringSerializer rowStringSerializer = new RowDataStringSerializer(leftType);
```

Comment:
as part of this change, let's move the row string serializer creation into open() -- make one for left and one for right rows, and use them consistently -- creating these for every printed row is a vestigial leftover from a prior refactor I didn't finish in join